Automattic/wp-calypso
# Stats: Improving chart labels and legend

Related to #75932, #76780

## Proposed Changes

* adding x-axis labels that are localised based on the user settings
* updating legend output based on selected period

## Testing Instructions



* Open the live branch for this PR
* Navigate to the traffic stats page at `/stats/day/:siteSlug`
* Add the query string `?flags=stats/subscribers-section` to enable the subscriber counts page
* Navigate to the `Subscribers` page and verify that it loads with a line chart
* Verify that the x-axis shows values appropriate for the selected period
   * For "Days", "Weeks" and "Months" shows "Apr 15"  and "Apr 13, 2023"
   * For "Years", shows "Jan 2023" and "Feb"
* Verify that the legend shows values appropriate for the selected period
   * For "Days", shows "May 10, 2023"
   * For "Weeks", shows "05/08/2023 - 05/14/2023"
   * For "Months", shows "May 2023"
   * For "Years", shows "2023"

## Pre-merge Checklist
